pandas/core/ops/__init__.py

```diff
@@ -309,7 +309,9 @@ def should_reindex_frame_op(
 
     if fill_value is None and level is None and axis is default_axis:
         # TODO: any other cases we should handle here?
-        cols = left.columns.intersection(right.columns)
+        # Only doing unique because of CategoricalIndex. Can be removed after
+        #  GH#38140 is merged
+        cols = left.columns.intersection(right.columns).unique()
```

Comment:
can you just do the intersection after L318?

Reply:
Way better, thank you very much.